p2p/src/secret_connection.rs

```diff
@@ -452,6 +496,23 @@ where
     }
 }
 
+impl<IoHandler: TryClone> TryClone for SecretConnection<IoHandler> {
```

Comment:
What is the advantage of using this trait as compared to holding an `Arc<IoHandler>` and `Clone`?

Reply:
See https://github.com/informalsystems/tendermint-rs/pull/938#discussion_r675794117